backend/space/collection.ts

```diff
@@ -0,0 +1,38 @@
+import type { HydratedDocument, Types } from "mongoose";
+import type { Space } from "./model";
+import SpaceModel from "./model";
+
+class SpaceCollection {
+    static async addOne(gmaps_place_payload: google.maps.places.PlaceResult): Promise<HydratedDocument<Space>> {
+        const place_id = gmaps_place_payload.place_id;
+        const formatted_address = gmaps_place_payload.formatted_address;
+        const formatted_phone_number = gmaps_place_payload.formatted_phone_number;
+        const name = gmaps_place_payload.name;
+        const photos = gmaps_place_payload.photos;
+        const url = gmaps_place_payload.url;
+        const website = gmaps_place_payload.website;
+        const space = new SpaceModel({
+            place_id,
+            formatted_address,
+            formatted_phone_number,
+            name,
+            photos,
+            url,
+            website
+        });
+        await space.save();
+        return space;
+    }
+
+    static async findOne(place_id: string): Promise<HydratedDocument<Space> | null>{
+        return SpaceModel.findOne({
+            place_id: place_id as string
+        });
+    }
+
+    static async deleteOne(place_id: string): Promise<void>{
```

Comment:
you're welcome to make these collection methods if you like, but I'd feel free to skip these ones like `findOne` and `deleteOne` that are essentially proxies for mongo queries directly, but it's up to you!